weave/ops_primitives/boolean.py

```diff
@@ -41,3 +41,11 @@ def none_coalesce(lhs: typing.Any, rhs: typing.Any):
     if isinstance(lhs, list) and isinstance(rhs, list):
         return [l or r for l, r in zip(lhs, rhs)]
     return lhs or rhs
+
+
+@op(output_type=lambda input_type: types.optional(input_type["results"].object_type))
```

Comment:
Back in the day, i built an `if` op. one perf issue was that it would execute both the true and false branches regardless of the truthiness of the incoming value. i think an alternative formulation could be:
```
if(cond:bool, else_fn: callable, else_fn: callable)
```

This would allow us to only execute the needed branch

Reply:
in this use case it is trivial, but could be more costly in others

Reply:
Might be worth a chat on this one.